doc/sphinx/src/outputs.rst

```diff
@@ -158,6 +161,134 @@ This will produce a text file (``.hst``) output file every 1 units of
 simulation time. The content of the file is determined by the functions
 enrolled by a specific package, see :ref:`state history output`.
 
+Histograms
+----------
+
+Parthenon supports calculating flexible 1D and 2D histograms in-situ that
+are written to disk in HDF5 format.
+Currently supported are
+
+- 1D and 2D histograms
+- binning by variable or coordinate (x1, x2, x3 and radial distance)
+- counting samples and or summing a variable  
+- weighting by volume and/or variable
+
+The output format follows ``numpy`` convention, so that plotting data
+with Python based machinery should be straightfoward (see example below).
+In general, histograms are calculated using inclusive left bin edges and
+data equal to the rightmost edge is also included in the last bin.
+
+A ``<parthenon/output*>`` block containing one simple and one complex
+example might look like::
+
+   <parthenon/output8>
+   file_type = histogram # required, sets the output type
+   dt = 1.0              # required, sets the output interval
+   num_histograms = 2    # required, specifies how many histograms are defined in this block
+
+   # 1D histogram ("standard", i.e., counting occurance in bin)
+   hist0_ndim = 1
+   hist0_x_variable = advected
+   hist0_x_variable_component = 0
+   hist0_x_edges_type = log
+   hist0_x_edges_num_bins = 10
+   hist0_x_edges_min = 1e-9
+   hist0_x_edges_max = 1e0
+   hist0_binned_variable = HIST_ONES
+
+   # 2D histogram of volume weighted variable according to two coordinates
+   hist1_ndim = 2
+   hist1_x_variable = HIST_COORD_X1
+   hist1_x_edges_type = list
+   hist1_x_edges_list = -0.5, -0.25, 0.0, 0.25, 0.5
+   hist1_y_variable = HIST_COORD_X2
+   hist1_y_edges_type = list
+   hist1_y_edges_list = -0.5, -0.1, 0.0, 0.1, 0.5
+   hist1_binned_variable = advected
+   hist1_binned_variable_component = 0
+   hist1_weight_by_volume = true
+   hist1_weight_variable = one_minus_advected_sq
+   hist1_weight_variable_component = 0
+
+with the following parameters
+
+- ``num_histograms=INT``
+   The number of histograms defined in this block.
+   All histogram definitions need to be prefix with ``hist#_`` where ``#`` is the
+   histogram number starting to count from ``0``.
+   All histograms will be written to the same output file with the "group" in the
+   output corresponding to the histogram number.
+- ``hist#_ndim=INT`` (either ``1`` or ``2``)
```

Comment:
maybe just `hist_dim`?

Reply:
I used `ndim` because it's also used in other places to specify/check for dimensionality.